cmd/xl-storage-format-v2.go

```diff
@@ -440,6 +440,9 @@ func (j xlMetaV2DeleteMarker) ToFileInfo(volume, path string) (FileInfo, error)
 		VersionID: versionID,
 		Deleted:   true,
 	}
+	if mv, ok := j.MetaSys[ReservedMetadataPrefixLower+"minio-version"]; ok {
+		fi.MinIOVersion = mv
+	}
```

Comment:
```suggestion
	if mv, ok := j.MetaSys[ReservedMetadataPrefixLower+"minio-version"]; ok {
		fi.MinIOVersion = mv
	}
```
This is 33 bytes longer.

An alternative could be:
```
type xlMetaV2Version struct {
	Type             VersionType           `json:"Type" msg:"Type"`
	ObjectV1         *xlMetaV1Object       `json:"V1Obj,omitempty" msg:"V1Obj,omitempty"`
	ObjectV2         *xlMetaV2Object       `json:"V2Obj,omitempty" msg:"V2Obj,omitempty"`
	DeleteMarker     *xlMetaV2DeleteMarker `json:"DelObj,omitempty" msg:"DelObj,omitempty"`
	WrittenByVersion []byte                `msg:"v,allownil"` // Written by MinIO version
}
```
This is 15 bytes, or even better:

```
	WrittenByVersion uint64                `msg:"v"` // Written by MinIO version
```
Would be 7 (since it will fit into uint32) bytes and faster since it doesn't require an extra alloc. 

Empty value would be `0`, which is easy to detect, and would be 3 bytes.

Reply:
(adding it is fully backcompat)

Reply:
Okay this is doable